### ISSUE TITLE
Doesn't work for self hosted github enterprise?

### ISSUE THREAD

**Issue by acastro2** [jdoiro3/mkdocs-multirepo-plugin#2]
I think it doesn't work for self-hosted github enterprise. The URL here is git.<company>.com (only vpn accessible), and when running mkdocs serve:

```bash
INFO     -  Multirepo plugin is importing docs for section Section 1
[Errno 2] No such file or directory: 'where'
```

This is an example of the mkdocs.yml:

```yaml
# Project information
site_name: Platform Wiki
site_description: Centralized location for all documentation
site_author: Platform

# Repository
repo_name: platform/centralized-docs
repo_url: https://git.<company>.com/platform/centralized-docs

nav:
  - Home: index.md
  - Login Server: '!import https://git.<company>.com/<org>/<repo>@master'

# Configuration
theme:
  name: material
  palette:
    primary: indigo
    accent: pink
    scheme: default

plugins:
  - multirepo:
      # (optional) tells multirepo to cleanup the temporary directory after site is built.
      cleanup: true
      # (optional) tells multirepo what the temp directory should be called
      temp_dir: multirepo_docs

```

**Comment by jdoiro3** [comment]
I'll need to look into this more deeply but it might be because I updated the `import` syntax. Instead of `@{branch}` it's now `?branch={branch}`. You can look at the README for more details. Let me know if you still encounter the same error. I'm planning on writing tests pretty soon.

**Comment by froschdesign** [comment]
@jdoiro3 
I have the same problem when I test the plugin on my local machine (macOS):

```bash
INFO     -  Building documentation...
INFO     -  Multirepo plugin importing docs...
[Errno 2] No such file or directory: 'where'
```

The problem is the detection of Git:

https://github.com/jdoiro3/mkdocs-multirepo-plugin/blob/8570df8de1132902a8101e6eb611d779118f3755/mkdocs_multirepo_plugin/util.py#L51-L56

This works for me:

```python
output = (
    subprocess.run('which git', capture_output=True, text=True, shell=True)
        .stdout
        .replace("\r", "")  # remove carrage return
        .split("\n")[0]  # handle multiple locations of git.exe
)

pprint(output) # /usr/local/git/bin/git
```

But the detection of the Git version is the next problem:

```bash
version = search(r'([\d.]+)', stdout).group(1).split(".")[:2]
AttributeError: 'NoneType' object has no attribute 'group'
```

https://github.com/jdoiro3/mkdocs-multirepo-plugin/blob/8570df8de1132902a8101e6eb611d779118f3755/mkdocs_multirepo_plugin/util.py#L82

**Comment by jdoiro3** [comment]
@froschdesign, thanks for looking into this. I'm pretty sure I know what's going on and how to resolve this issue. The below function only checks for Linux and not macOS, leading to *where git* being called which isn't a macOS command.

```python
def execute_bash_script(script: str, arguments: list = [], cwd: Path = Path.cwd()) -> subprocess.CompletedProcess:
    """executes a bash script"""
    extra_run_args = get_subprocess_run_extra_args()
    if platform == "linux" or platform == "linux2":
        process = subprocess.run(
            ["bash", script]+arguments, cwd=cwd, **extra_run_args
        )
    else:
        git_folder = where_git()
        process = subprocess.run(
            [str(git_folder / "bin" / "bash.exe"), script]+arguments, cwd=cwd, **extra_run_args
        )
    return process
```

Also - that's 100% a good idea but didn't go with GitPython b/c from what I saw it didn't support the advance features like sparse clone or the *--depth* argument, which are optimizations.

**Comment by jdoiro3** [comment]
I just bumped changes that *should* have fixed this bug in *v0.2.8*. Can one of you confirm for me?

**Comment by froschdesign** [comment]
@jdoiro3 
Works great! 👍
